src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -186,7 +186,26 @@ class ADRController(private val encryption: Encryption,
     fun createRelease(@PathVariable id: String, @RequestParam name: String): ResponseEntity<String>
     {
         val adr = adrClientBuilder.build()
-        return adr.post("dataset_version_create", listOf("dataset_id" to id, "name" to name));
+        var error: ResponseEntity<String>? = null
+        val releasesResponse = adr.get("/dataset_version_list?dataset_id=${id}")
+        if (releasesResponse.statusCode == HttpStatus.OK) {
+            val releases = objectMapper.readTree(releasesResponse.body!!)["data"]
+            val duplicateRelease = releases.find { it["name"].asText() == name }
+            if (duplicateRelease != null) {
+                val duplicateReleaseId = duplicateRelease["id"].asText()
+                val deleteResponse = adr.post("/version_delete", listOf("version_id" to duplicateReleaseId));
+                if (deleteResponse.statusCode != HttpStatus.OK){
+                    error = deleteResponse
+                }
+            }
+        } else {
+            error = releasesResponse
+        }
+        if (error != null){
+            return error
+        } else {
+            return adr.post("/dataset_version_create", listOf("dataset_id" to id, "name" to name));
+        }
     }
```

Comment:
@EmmaLRussell is this the kind of error handling that is expected? For example, is it right that if we get an error trying to get the existing releases or deleting a release, should we arrest the attempt to create a release and return the offending error? Or might that be confusing for the user if they get a failed to delete release error message after trying to create a release? What do you think the appropriate behaviour should be?